### PR TITLE
Few changes suggested

### PR DESCRIPTION
**Pull request by ares**
let's discuss changes here, I'll update PR as needed

### CODE REVIEW

File: pages/source/documentation/index.md [hunk 1]
```diff
@@ -86,8 +87,50 @@ Dynflow has been developed to be able to support orchestration of services in th
 
 ### Action anatomy
 
-Each action can be viewed as a function. It's planned for execution with 
-input, then it's executed producing output quite possibly having side-effects. 
+When action is triggered, Dynflow executes plan method on this action, which
+is responsible to build the plan. It builds the plan by calling plan_action
```
Comment (adamruzicka) [inline pages/source/documentation/index.md]: Shouldn't this be something like "which is responsible for building the plan"?

File: pages/source/documentation/index.md [hunk 2]
```diff
@@ -688,8 +737,8 @@ for more details.
 Each **Action** phase can be in one of the following states:
 
 -   **Pending** - Not yet executed.
--   **Running** - Action phase is being executed right now.
--   **Success** - Action phase execution finished successfully.
+-   **Running** - Action phase ?which? is being executed right now.
+-   **Success** - Action phase ?which? execution finished successfully.
```
Comment (adamruzicka) [inline pages/source/documentation/index.md]: Doesn't it sound more natural this way?
Action phase whose execution finished successfully.
Action phase which successfully finished executing.

File: pages/source/documentation/index.md [hunk 3]
```diff
@@ -86,8 +87,50 @@ Dynflow has been developed to be able to support orchestration of services in th
 
 ### Action anatomy
 
-Each action can be viewed as a function. It's planned for execution with 
-input, then it's executed producing output quite possibly having side-effects. 
+When action is triggered, Dynflow executes plan method on this action, which
+is responsible to build the plan. It builds the plan by calling plan_action
+and plan_self methods, effectively listing actions that should be run as 
+a part of this plan. In other words, it compiles a list of actions on which 
+method run will be called. Also it's responsible for giving these actions
```
Comment (adamruzicka) [inline pages/source/documentation/index.md]: Maybe mention ```finalize``` here?
on which ```run``` and later ```finalize``` methods will be called?

File: pages/source/documentation/index.md (hunk 2 above)
Comment (ares) [inline pages/source/documentation/index.md]: the ?which? was actually my question, not suggestion :-) I think we can remove these now

File: pages/source/documentation/index.md (hunk 1 above)
Comment (ares) [inline pages/source/documentation/index.md]: right, fixing

File: pages/source/documentation/index.md (hunk 3 above)
Comment (ares) [inline pages/source/documentation/index.md]: Does plan method deals with finalize anyhow? I thought it's isolated and is triggered only if it's defined for the action.

**Comment by pitr-ch** [comment]
Thanks @ares  and @adamruzicka. I'll merge it with my updates, closing this one when it's done.

**Comment by pitr-ch** [comment]
Please do not push more changes here, let me know instead, thanks.

**Comment by pitr-ch** [comment]
Pulled to https://github.com/Dynflow/dynflow/pull/138

**Pull request closed without merging**